packages/core/bootstrap/src/lib/ws/epics.ts

```diff
@@ -297,28 +365,23 @@ export const connectEpic: Epic<AnyAction, AnyAction, { ws: RootState }, any> = (
               withLatestFrom(state$),
               mergeMap(([message, state]) => {
                 const isActiveSubscription = !!state.ws.subscriptions.all[subscriptionKey]?.active
+                const actionPayload = {
+                  message,
+                  subscriptionKey,
+                  input: payload.input,
+                  context: payload.context,
+                  connectionInfo: payload.connectionInfo,
+                  wsHandler,
+                }
                 if (!isActiveSubscription) {
                   logger.info('WS: Subscribed', subscriptionMeta(payload))
                   return of(
                     subscribeFulfilled(payload),
-                    messageReceived({
-                      message,
-                      subscriptionKey,
-                      input: payload.input,
-                      context: payload.context,
-                      connectionInfo: payload.connectionInfo,
-                    }),
+                    writeToCache(actionPayload),
```

Comment:
Do we need a whole new action for `writeToCache`? Could we just use `messageRecieved` in the epic instead?

`messageRecieved` currently has no use besides in the metrics epic.

Reply:
Yeah I think we can do that